### PR TITLE
[MM-48825] "plugin configured with a nil SecureConfig" warning logged when starting each plugin

### PR DESCRIPTION
**Pull request by javaguirre**
#### Summary

go-plugin maintainers added a new warning for when there is [no SecureConfig struct configured](https://github.com/hashicorp/go-plugin/pull/207) in the ClientConfig for the plugin connection, this PR removes that warning.

#### Release Note

```release-note
Fix [MM-48825] "plugin configured with a nil SecureConfig" warning logged when starting each plugin
```

[MM-48825]: https://mattermost.atlassian.net/browse/MM-48825?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: plugin/supervisor.go [hunk 1]
```diff
@@ -4,7 +4,10 @@
 package plugin
 
 import (
+	"crypto/md5"
```
Comment (hanzei) [inline plugin/supervisor.go:7]: Why was `md5` picked here over other hash algorithms?  Does it matter that md5 is cryptographically broken?

**Review by hanzei — CHANGES_REQUESTED** [review]

File: plugin/supervisor.go (hunk 1 above)
Comment (javaguirre) [reply plugin/supervisor.go:7]: I don't mind using another.

Md5 is less costly in terms of performance and an attacker would need to generate a binary of their *evil* code that could generate the same hash as the plugin's executable file.

Comment (javaguirre) [reply plugin/supervisor.go]: switched to sha256, thanks Ben!

**Review by mickmister — APPROVED** [review]

**Review by hanzei — CHANGES_REQUESTED** [review]
Should the client i.e. the plugin also check these checksum? In other words: Is this change supposed to be an actually security improvement?

**Comment by javaguirre** [comment]
> Should the client i.e. the plugin also check these checksum? In other words: Is this change supposed to be an actually security improvement?

The first use case I could think of for this would be having a Registry of Apps checksums (in a Marketplace) for example and compare the checksum stored in the server with the checksum of the executable we are trying to install at a certain time.

I'll create a ticket for that if we want to discuss it, but that's beyond the scope of this PR, Thank you @hanzei !

**Comment by hanzei** [comment]
I'm wondering if we should make it clear that these checksums are not used right now. As a reader of the code, it should be clear that it's not a security feature (right now).

**Comment by javaguirre** [comment]
> I'm wondering if we should make it clear that these checksums are not used right now. As a reader of the code, it should be clear that it's not a security feature (right now).

I'll add a comment then.

File: plugin/supervisor.go [hunk 2]
```diff
@@ -62,6 +65,14 @@ func newSupervisor(pluginInfo *model.BundleInfo, apiImpl API, driver Driver, par
 
 	cmd := exec.Command(executable)
 
+	// This doesn't add more security than before
+	// but removes the SecureConfig is nil warning
```
Comment (hanzei) [inline plugin/supervisor.go:69]:
```suggestion
	// but removes the SecureConfig is nil warning.
```

**Review by hanzei — APPROVED** [review]

**Comment by hanzei** [comment]
@DHaussermann Can we merge this PR with QA review as it's only about a log line

**Review by DHaussermann — APPROVED** [review]
Tested and passed:
- The nil security line no longer appaers to be written to the log
- Tested including `DEBUG` as log level
- Tested on HA just in vase behavior is different
- No functional impact found  

LGTM!

Sorry for the delay on this.
Please merge.

**Pull request merged**